lib/ansible/plugins/lookup/onepassword.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2018, Scott Buchanan <sbuchanan@ri.pn>
+# Copyright (c) 2016, Andrew Zenk <azenk@umn.edu> (lastpass.py used as starting point)
+# Copyright (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+    lookup: onepassword
+    author:
+      -  Scott Buchanan <sbuchanan@ri.pn>
+      -  Andrew Zenk <azenk@umn.edu>
+    version_added: "2.6"
+    requirements:
+      - op (1Password command line utility: https://support.1password.com/command-line/)
+      - must have already logged into 1Password using op CLI
+    short_description: fetch data from 1Password
+    description:
+      - use the op command line utility to fetch specific fields from 1Password
+    options:
+      _terms:
+        description: identifier(s) of object(s) from which you want to retrieve the field (UUID, name or domain)
+        required: True
+      field:
+        description: field to return from 1Password
+        default: 'password'
+      vault:
+        description: vault to retrieve from (if absent will search all available vaults)
+        default: None
+"""
+
+EXAMPLES = """
+- name: "retrieve password from entry named KITT in any vault"
+  debug: password="{{ lookup('onepassword', 'KITT') }}"
+
+- name: "retrieve username from entry named HAL 9000 in vault Discovery"
+  debug: password="{{ lookup('onepassword', 'HAL 9000', field='username', vault='Discovery') }}"
+"""
+
+RETURN = """
+  _raw:
+    description: field data requested
+"""
+
+import json
+from subprocess import Popen, PIPE
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+
+class OnePassException(AnsibleError):
+
+    pass
+
+
+class OnePass(object):
+
+    def __init__(self, path='op'):
+        self._cli_path = path
+
+    @property
+    def cli_path(self):
+        return self._cli_path
+
+    def assert_logged_in(self):
+        try:
+            self._run(["get", "account"])
+        except:
+            raise OnePassException("Not logged into 1Password: please run 'op signin' first")
+
+    def get_field(self, item_id, field, vault=None):
+        args = ["get", "item", item_id]
+        if vault is not None:
+            args += ['--vault={0}'.format(vault)]
+        output, dummy = self._run(args)
+        return self._parse_field(field, output) if output != '' else ''
+
+    def _run(self, args, stdin=None, expected_rc=0):
```

Comment:
Likewise, this was copied from the `lastpass` plugin that I used as a base for this. Happy to change, though.

Reply:
I went to implement this, but then I saw that `run_cmd` requires the command to be passed as an assembled string, only to be broken apart again by `shlex`. To pass the command as a string it seems I'd have to take on the burden of escaping the item ID and vault ID rather than allowing the `subprocess` library to do it for me.

At least in the current version, it doesn't look like `run_cmd` provides any additional safety/compatibility not already provided in the method here. The one potential issue would be the conversion of `stdin` to a byte string, but that's not even used here. I'm going ahead and removing this parameter to avoid potential issues in the future.

Given all this, are you OK with leaving my run method?

Reply:
Yes, that's fine.